### ISSUE TITLE
Use set for array union and array remove

### ISSUE THREAD

**Issue by jakeleventhal** [googleapis/nodejs-firestore#1248]
**Is your feature request related to a problem? Please describe.**
It would be nice to have array union and array remove for javascript `Set`s
 **Describe the solution you'd like**
Instead of:
```javascript
const s = new Set();
s.add('a');
s.add('b');
Firestore.FieldValue.arrayUnion(Array.from(s));
```
it would be nice to be able to just do:
```javascript
const s = new Set();
s.add('a');
s.add('b');
Firestore.FieldValue.arrayUnion(s);
```

**Comment by schmidt-sebastian** [comment]
@jakeleventhal I will discuss this with our team. `arrayUnion()` currently takes varargs, which would make the resulting API a tiny bit obnoxious. I should have an answer for you by end the of next week.

**Comment by jakeleventhal** [comment]
@schmidt-sebastian thank you!

**Comment by schmidt-sebastian** [comment]
I have talked to my team about this and we decided against this for now. This is mainly because we are currently taking varargs, and special-casing Set as an argument is strange when Arrays are explicitly not supported.  At some point, we also do want explore the option of allowing arrays within arrays, which would then mean that `arrayUnion(Set<>)` would have to act very different from `arrayUnion([])`.
